apps/04TowersOfHanoi.js

```diff
@@ -7,43 +7,81 @@ var rl = readline.createInterface({
   output: process.stdout
 });
 
+//this is just the board
 var stacks = {
-  a: [4, 3, 2, 1],
-  b: [],
-  c: []
+  tower1: [4, 3, 2, 1],
+  tower2: [],
+  tower3: []
 };
 
 function printStacks() {
-  console.log("a: " + stacks.a);
-  console.log("b: " + stacks.b);
-  console.log("c: " + stacks.c);
+  console.log("tower1: " + stacks.tower1);
+  console.log("tower2: " + stacks.tower2);
+  console.log("tower3: " + stacks.tower3);
 }
 
-function movePiece() {
+function movePiece(startMove, endMove) {
   // Your code here
+    var choice = stacks[startMove];
+    var endChoice = stacks[endMove];
+    var block = choice.pop();
 
+    endChoice.push(block);
 }
 
-function isLegal() {
+function isLegal(startMove, endMove) {
   // Your code here
+  var choice = stacks[startMove];
+  var endChoice = stacks[endMove];
+  var block = choice[choice.length -1];
+  var block2 = endChoice[endChoice.length -1];
+
+  if( endChoice.length === 0){
+    return true;
+  }
+
+  if(block > block2){
+    return false;
+  }
+
+  return true;
 
 }
 
-function checkForWin() {
+function checkForWin(startMove, endMove) {
   // Your code here
+  var choice = stacks[startMove];
+  var endChoice = stacks[endMove];
+
+  // does stack have a length of 4
+  if(stacks.tower2.length === 4 || stacks.tower3.length === 4){
+    return true;
+  }
+  return false;
+
 
 }
 
-function towersOfHanoi(startStack, endStack) {
+function towersOfHanoi(startMove, endMove) {
   // Your code here
+  movePiece(startMove, endMove);
+  isLegal(startMove, endMove); //if true make the move
```

Comment:
Your movePiece() function and isLegal() function are independent of each other. Therefore, your code will force a move before it checks to see if its legal.

Use an `if() {..}` statement when calling movePiece.